### ISSUE TITLE
Invalid model for QF_NIA formulas (theory_engine.cpp:2283) (check-models)

### ISSUE THREAD

**Issue by rainoftime** [CVC4/CVC4#4464]
Hi,
For the following formula:
~~~~
(set-logic QF_NIA)
(declare-const i2 Int)
(push 1)
(assert false)
(check-sat)
(pop 1)
(check-sat)
(check-sat)
(push 1)
(assert (xor true true true true (>= 43 (div i2 38)) true true true true true))
(push 1)
(pop 1)
(pop 1)
(assert (xor false true (= 0 (div i2 38)) true true true true))
(check-sat)
~~~~

CVC4 throws out a fatal failure:
~~~~
unsat
sat
sat
Fatal failure within void CVC4::TheoryEngine::checkTheoryAssertionsWithModel(bool) at /home/peisen/test/tofuzz/CVC4/src/theory/theory_engine.cpp:2283
Internal error detectedTHEORY_ARITH has an asserted fact that the model doesn't satisfy.
The fact: (not (= (div i2 38) 0))
Model value: false

Aborted (core dumped)
~~~~

OS: Ubuntu 16.04
Commit: ad7907adf

**Comment by rainoftime** [comment]
~~~~
(set-logic QF_NIA)
(declare-const i11 Int)
(check-sat)
(push 1)
(assert (distinct (mod i11 698) 0))
(push 1)
(pop 1)
(assert false)
(check-sat)
(check-sat)
(pop 1)
(check-sat)
(assert (= true true true (distinct (mod i11 698) 0) true))
(check-sat)
~~~~

**Comment by rainoftime** [comment]
~~~~
(set-logic QF_NIA)
(declare-fun _substvar_335_ () Int)
(declare-fun _substvar_417_ () Int)
(check-sat)
(declare-const i3 Int)
(check-sat)
(check-sat)
(push 1)
(assert (xor true true true true true true (= (mod i3 54) _substvar_417_) false true true true))
(push 1)
(assert false)
(push 1)
(check-sat)
(pop 1)
(pop 1)
(check-sat)
(push 1)
(assert false)
(check-sat)
(check-sat)
(pop 1)
(check-sat)
(check-sat)
(check-sat)
(check-sat)
(pop 1)
(assert (>= (mod (mod i3 54) _substvar_335_) 0 0))
(check-sat)

~~~~

**Comment by 4tXJ7f** [comment]
I've confirmed that the issues on the first two inputs have been fixed with 52082c72d78eee219e3049285d5df559dacac8b5. The issue in the last input was first fixed/masked with 927066eaecfc2c6f00aa1aca695b68e70164aae3 but could still be triggered with `--debug-check-model`. 52082c72d78eee219e3049285d5df559dacac8b5 fixes the issue on the last input with `--debug-check-model`.
